src/sqlfluff/core/rules/base.py

```diff
@@ -1103,12 +1145,19 @@ def get_rulelist(self, config) -> List[BaseRule]:
         """
         # Validate all generic rule configs
         self._validate_config_options(config)
+        # Find all valid groups for ruleset
+        valid_groups = set(
+            [group for _, attrs in self._register.items() for group in attrs["groups"]]
+        )
         # default the allowlist to all the rules if not set
         allowlist = config.get("rule_allowlist") or list(self._register.keys())
         denylist = config.get("rule_denylist") or []
 
         allowlisted_unknown_rule_codes = [
-            r for r in allowlist if not fnmatch.filter(self._register, r)
+            r
+            for r in allowlist
+            # Add valid groups to the register when searching for invalid rules _only_
+            if not fnmatch.filter({**self._register, **dict.fromkeys(valid_groups)}, r)
```

Comment:
I'm having a pretty hard time understanding this line of code. Is there a way to write it that's more clear?

Reply:
Previously `fnmatch.filter` was being used on the register dict, meaning it had all the rules as the keys of the dict to search through valid rule codes. I wanted to add all the valid groups as keys to that dict

Perhaps a better way would be to just create a separate list of rule codes from the register dict keys and then add these valid groups to that list. How about that?

Reply:
That sounds good. To be clear, the existing code is legit, just a little too terse for my brain to process, at least on a Friday evening. 😅